library/alloc/src/raw_vec.rs

```diff
@@ -389,6 +389,9 @@ impl<T, A: AllocRef> RawVec<T, A> {
     // so that all of the code that depends on `T` is within it, while as much
     // of the code that doesn't depend on `T` as possible is in functions that
     // are non-generic over `T`.
+    //
+    // We never inline this function to keep `Vec::push` small.
+    #[inline(never)]
```

Comment:
In many places, `#[cold]` is used for this, sometimes together with `#[inline(never)]` on the same function. Is there some guidance somewhere on which to use where?

If these kind of functions should have `#[inline(never)]` instead (or in addition to) of `#[cold]`, some of them (e.g. `SyncOnceCell::initialize`) should probably be updated.

Reply:
hmm, I am not really sure. It probably depends on where it is used and we would need an additional perf run :laughing: 

I do feel like this method is kind of special as it is extremely frequently used. Not sure if `SyncOnceCell` is used frequently enough for this to be noticeable. Could be worth trying though